### PR TITLE
Theme-style refactor

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
This includes the theme-style refactor and implementation for the intro step and shared big image steps. RSDStepViewController needs some serious love... next week. :)

### CODE REVIEW

File: CRFModuleValidation/Assets.xcassets/closeActivity.imageset/Contents.json [hunk 1]
```diff
@@ -10,7 +10,6 @@
     "author" : "xcode"
   },
   "properties" : {
-    "template-rendering-intent" : "template",
-    "preserves-vector-representation" : true
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Assets.xcassets/closeActivity.imageset/Contents.json:14]: we don't want this for this one?

File: CRFModuleValidation/CRFTaskFactory.swift [hunk 2]
```diff
@@ -35,43 +35,4 @@ import ResearchSuite
 
 open class CRFTaskFactory: RSDFactory {
     
-    public enum CRFStepType: String {
```
Comment (Erin-Mounts) [inline CRFModuleValidation/CRFTaskFactory.swift:38]: Is this class definition now empty? Do we still need it?

File: CRFModuleValidation/Cardio_12MT.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<link href="../Learn_section.css" rel="stylesheet" type="text/css" />
+</head>
+
+<body>
+
+<div class="container">
+    <div class="content">
+        <h1>About 12 minute test</h1>
+        <h2>Why this test</h2>
+        <p>Recording a 12 minute  test is an important measure of your overall health. It is associated with overall mortality and many common diseases in the US such as cardiovascular disease and Alzheimer’s Disease.</p>
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Cardio_12MT.html]: extra space between "minute" and "test"

File: CRFModuleValidation/Cardio_12MT.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        <h2>What to do</h2>
+        <ol>
+            <li>Measure your <strong>heart rate</strong> within this app before running for 12 minutes.</li>
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Cardio_12MT.html]: since "after" is part of the bolded text in the last list item, should "before" be bolded here?

File: CRFModuleValidation/Cardio_12MT.json [hunk 5]
```diff
@@ -21,33 +40,42 @@
...
                            {
                            "identifier"   : "volumeUp",
                            "type"         : "instruction",
                            "title"        : "Volume turned up?",
-                           "imageBefore"  : "volumeCheckIcon",
                            "text"         : "Bring your phone with you and turn up your phone volume so you can hear the instructions while you are moving.",
-                           "actions"           : { "goForward": { "buttonTitle": "It’s turned on" }}
+                           "image"        : { "imageName": "volumeCheckIcon", "placementType": "topBackground" },
+                           "colorTheme"   : { "backgroundColor" : "perrywinkle", "usesLightStyle" : true },
+                           "actions"      : { "goForward": { "buttonTitle": "It’s turned on" }},
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Cardio_12MT.json:56]: should it say "It's turned up" ?

File: CRFModuleValidation/Cardio_12MT.json [hunk 6]
```diff
@@ -21,33 +40,42 @@
...
+                           "viewTheme"    : { "viewIdentifier": "Instruction", "storyboardIdentifier": "ActiveTaskSteps" }
                            },
                            {
                            "identifier"   : "goOutside",
                            "type"         : "instruction",
                            "title"        : "Go outside",
-                           "imageBefore"  : "goOutsideIcon",
                            "text"         : "Start by going oustide and standing still to measure your resting heart rate before you start moving.",
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Cardio_12MT.json:63]: Seems like we should be having them stand still for a certain amount of time before they even start taking their resting heart rate. Just walking outside could bump it up significantly, and what if they had to climb stairs or go up a hill to get out to where they're starting their run? (Also shouldn't a resting heart rate generally be taken sitting down?)

These are probably questions better posed to the study designers, but they're not making this pull request, you are... ;)

File: CRFModuleValidation/Cardio_12MT.json [hunk 7]
```diff
@@ -57,41 +85,45 @@
...
                            "identifier"         : "heartRate.after",
-                           "type"               : "heartRate",
-                           "title"              : "Stand still for 1 minute",
-                           "subtitle"           : "Almost done! Stand still for a minute to measure your heart rate recovery.",
-                           "detail"             : "Your heart rate changed to",
-                           "taskTransformer"    : { "resourceName": "HeartrateStep.json"}
+                           "type"               : "transform",
+                           "replacementSteps"   : [{ "identifier"   : "instruction",
+                                                   "title"        : "Stand still for 1 minute",
+                                                   "text"         : "Almost done! Stand still for a minute to measure your heart rate recovery." },
```
Comment (Erin-Mounts) [inline CRFModuleValidation/Cardio_12MT.json:122]: Are they supposed to be doing the finger-over-camera-and-flash thingy again here? If so, the instructions should maybe mention it? If not, where are we getting the heart rate from for the feedback?

File: CRFModuleValidation/HeartrateStep.json [hunk 8]
```diff
@@ -3,22 +3,23 @@
...
-                           "viewInfo"       : { "viewIdentifier": "ActiveInstruction", "storyboardIdentifier": "ActiveTaskSteps" }
+                           "viewTheme"      : { "viewIdentifier": "ActiveInstruction", "storyboardIdentifier": "ActiveTaskSteps" }
                            },
                            {
                            "identifier"   : "heartRate",
                            "type"         : "active",
                            "text"         : "Gently cover both the camera and flash with your finger.",
                            "shouldHideActions": ["goBackward", "goForward"],
-                           "duration": 60,
+                           "duration": 6,
```
Comment (Erin-Mounts) [inline CRFModuleValidation/HeartrateStep.json:22]: Is this change intentional, or did you do it to speed up testing and then forgot to change it back?

**Review by Erin-Mounts — APPROVED** [review]
Nits and a few questions, otherwise looks good, merge when ready

File: CRFModuleValidation/CRFTaskFactory.swift (hunk 2 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/CRFTaskFactory.swift:38]: We will need it for the recorders so I left it stubbed in so I could leave the override in place.

File: CRFModuleValidation/Assets.xcassets/closeActivity.imageset/Contents.json (hunk 1 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/Assets.xcassets/closeActivity.imageset/Contents.json:14]: I realized that I was turning it on for everything, including assets that don't resize, except when they are added to the bundle. Close button, back button, info button are all 32x32 and are independent of device.

File: CRFModuleValidation/Cardio_12MT.html (hunk 4 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/Cardio_12MT.html]: No idea why they picked those words to be strong. It looks kind of weird and I'm guessing will need to be word-smithed. I'm guessing that they are trying to highlight why they take your heart rate twice.

File: CRFModuleValidation/Cardio_12MT.json (hunk 5 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/Cardio_12MT.json:56]: Maybe? Personally, I like the fitbit button best. "It's on" ;)

File: CRFModuleValidation/Cardio_12MT.json (hunk 6 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/Cardio_12MT.json:63]: Yup. I've asked. And I've pointed out why I was gathering location data from the start of the task, but they don't want it. *shrug* I also wanted to put the volume check after the go outside step to give the user a bit more time for their heart rate to come to resting.

File: CRFModuleValidation/Cardio_12MT.json (hunk 7 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/Cardio_12MT.json:122]: The instructions are also included in the actual heart rate step... but not yet. (cuz I haven't done that step yet).

File: CRFModuleValidation/HeartrateStep.json (hunk 8 above)
Comment (syoung-smallwisdom) [reply CRFModuleValidation/HeartrateStep.json:22]: Yes, this is intentional. Thanks for double-checking though.  :)  (The 12 minute run is also intentionally 12 seconds.)
